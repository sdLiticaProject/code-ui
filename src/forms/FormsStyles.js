import styled, {keyframes} from "styled-components";

const InputFormWrapper = styled.div`
  position: relative;
  padding: 15px 0 0;
  margin-top: 5px;
  width: 100%;

  & input {
  font-family: sans-serif;
  width: 70%;
  border: 0;
  border-bottom: 2px solid #9b9b9b;
  outline: 0;
  font-size: 1.1rem;
  color: #9b9b9b;
  padding: 7px 0;
  background: transparent;
  transition: border-color 0.2s;

  &::placeholder {
    color: red;
  }

  &:focus {
    &::placeholder {
      color: transparent;
    }
    padding-bottom: 6px;  
    border-width: 3px;
    border-image: linear-gradient(to right, #2196f3,#bbdefb);
    border-image-slice: 1;
  }

  &:required,&:invalid { 
    box-shadow:none;
   }
  }

  & label {
    font-size: 1rem;
    color: #9b9b9b;
    font-family: sans-serif;
    position:absolute;
    bottom:40px;
  }
`;
const Button = styled.button`
  background-color: #87c6f8;
  border: 1px solid #FFFFFF;
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.075) inset;
  transition: border 0.2s linear 0s, box-shadow 0.2s linear 0s;
  border-radius: 6px;
  color: white;
  display:block;
  width:230px;
  margin: 5% auto;
  font-family: sans-serif;
  font-size: 18px;
  text-align:center;
  height: 40px;
  line-height: 20px;
  margin-bottom: 10px;
  padding: 4px 6px;
  vertical-align: middle;
  text-decoration:none;

  &:hover,&:focus {
    border-color: white;
    box-shadow: 0 1px 1px rgba(0, 0, 0, 0.075) inset, 0 0 8px rgba(82, 168, 236, 0.3);
    outline: 0 none;
  }
`;


const Error = styled.text`
  color: #8B0000;
  font-size:0.8rem;
  font-family:sans-serife;
  text-decoration:none;
  position:absolute;
  right:0px;
  top:10px;
  padding-right:67px;

  @media (max-width: 500px) {
    padding-right: 60px;
}
`;

const Link = styled.a`
  color: #9b9b9b;
  text-decoration:none;
  position:absolute;
  padding-right: 67px;
  right: 0;
  font-family:sans-serife;

  @media (max-width: 500px) {
    padding-right: 60px;
}
`;

const Transfer = styled.a`
  position:absolute;
  left:176px;
  color: #3eacdc;
  text-decoration:none;
  font-family:sans-serife;

  @media (max-width: 500px) {
    left: 150px;
}
`;

export { Button,  InputFormWrapper, Link, Error, Transfer };
