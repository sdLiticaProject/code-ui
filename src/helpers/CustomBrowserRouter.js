import React from 'react';
import {BrowserRouter, Route} from 'react-router-dom';
import PropTypes from 'prop-types';

export const RouterContext = React.createContext({});

const CustomBrowserRouter = ({children}) => (
  <BrowserRouter>
    <Route>
      {routeProps => <RouterContext.Provider value={routeProps}>{children}</RouterContext.Provider>}
    </Route>
  </BrowserRouter>
);

CustomBrowserRouter.propTypes = {
  children: PropTypes.object.isRequired
};

export default CustomBrowserRouter;
