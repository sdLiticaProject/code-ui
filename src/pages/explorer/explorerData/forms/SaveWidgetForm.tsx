import React, {useState} from "react";
import {useSnackbar} from "notistack";
import axios from "axios";
import * as api from "../../../../constants/api";
import Cookies from "js-cookie/src/js.cookie";
import WidgetForm from "./WidgetForm";
import {Button} from "@material-ui/core";

const SaveWidgetForm = (): JSX.Element => {
    const {enqueueSnackbar, closeSnackbar} = useSnackbar()
    const showWidgetModal = () => {
        setIsWidgetModalOpen(true)
    }
    const handleUpdateBucketOk = () => {
        setIsWidgetModalOpen(false)
    }
    const handleWidgetCancel = () => {
        setIsWidgetModalOpen(false)
    }
    const [initialValues, setInitialValues] = useState(undefined);
    const [isWidgetModalOpen, setIsWidgetModalOpen] = useState(false);

    const onFormSubmit = async (e) => {
        console.log("widget-data:", e)
        handleUpdateBucketOk()
        enqueueSnackbar("Widget saved successfully ", {
            autoHideDuration: 2000,
            variant: "success"
        })

        // const requestBody = {
        //     "name": e.name,
        //     "description": e.description,
        //     "retentionPeriod": e.retention !== 'custom' ? convertRetentionToSeconds(e.retention) : convertRetentionToSeconds(e.retentionPeriod)
        // }

        // try {
        //     const response = await axios.post(api.bucketList() + `/${bucketId}`, requestBody,{
        //         headers: {Authorization: `cloudToken ${Cookies.get('token')}`},
        //     });
        //     enqueueSnackbar("Bucket update successfully ", {
        //         autoHideDuration: 5000,
        //         variant: "success"
        //     })
        //     handleUpdateBucketOk()
        // } catch (error) {
        //     enqueueSnackbar("Some problems", {
        //         autoHideDuration: 5000,
        //         variant: "error"
        //     })
        // }
    }


    return (
        <>
            <Button variant="contained" onClick={showWidgetModal}>
                Save As
            </Button>
            <WidgetForm formVisible={isWidgetModalOpen}
                        onCancel={handleWidgetCancel}
                        onFinish={onFormSubmit}
            />
        </>
    )
}

export default SaveWidgetForm;