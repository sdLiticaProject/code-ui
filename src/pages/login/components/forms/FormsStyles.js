import styled, { keyframes } from 'styled-components';

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

export const FormWrapper = styled.form`
  display: flex;
  flex-direction: column;
`;

export const NavLinkWrapper = styled.div`
  margin-bottom: 30px;
`;

export const LoadIndicator = styled.div`
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  margin: 15px;
  display: inline-block;
  font-size: 10px;
  text-indent: -9999em;
  border-top: 3px solid #444;
  border-right: 3px solid #444;
  border-bottom: 3px solid #444;
  border-left: 3px solid #00b7db;
  animation: ${rotate} 0.8s infinite linear;
  border-top-color: #e4e9eb;
  border-right-color: #e4e9eb;
  border-bottom-color: #e4e9eb;
  border-left-color: #87c6f8;
  border-radius: 50%;
  width: 10px;
  height: 10px;
  position: absolute;
  top: 0px;
  z-index: 99;
  margin-top: 20px;
  margin-left: 20px;

  ${({ tab }) => (tab === '/news' || tab === 'unknown') && `border-left-color: green !important;`};
`;

const InputFormWrapper = styled.div`
  position: relative;
  padding: 15px 0 0;
  margin-top: 5px;
  width: 100%;

  & input {
    font-family: sans-serif;
    width: 70%;
    border: 0;
    border-bottom: 2px solid #9b9b9b;
    outline: 0;
    font-size: 1.1rem;
    color: #9b9b9b;
    padding: 7px 0;
    background: transparent;
    transition: border-color 0.2s;

    &::placeholder {
      color: red;
    }

    &:focus {
      &::placeholder {
        color: transparent;
      }
      padding-bottom: 6px;
      border-width: 3px;
      border-image: linear-gradient(to right, #2196f3, #bbdefb);
      border-image-slice: 1;
    }

    &:required,
    &:invalid {
      box-shadow: none;
    }
  }

  & label {
    font-size: 1rem;
    color: #9b9b9b;
    font-family: sans-serif;
    position: absolute;
    bottom: 40px;
  }
`;

const Button = styled.button`
  background-color: #87c6f8;
  cursor: pointer;
  user-select: none;
  border: 1px solid #ffffff;
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.075) inset;
  transition: border 0.2s linear 0s, box-shadow 0.2s linear 0s;
  border-radius: 6px;
  color: white;
  display: block;
  width: 230px;
  margin: 5% auto;
  font-family: sans-serif;
  font-size: 18px;
  text-align: center;
  height: 40px;
  line-height: 20px;
  margin-bottom: 10px;
  padding: 4px 6px;
  text-decoration: none;
  &:hover,
  &:focus {
    border-color: white;
    box-shadow: 0 1px 1px rgba(0, 0, 0, 0.075) inset, 0 0 8px rgba(82, 168, 236, 0.3);
    outline: 0 none;
  }
`;

const Error = styled.text`
  color: #8b0000;
  font-size: 0.8rem;
  font-family: sans-serife;
  text-decoration: none;
  position: absolute;
  right: 0px;
  top: 10px;
  padding-right: 67px;

  @media (max-width: 500px) {
    padding-right: 60px;
  }
`;

const Link = styled.a`
  color: #9b9b9b;
  text-decoration: none;
  position: absolute;
  padding-right: 67px;
  right: 0;
  font-family: sans-serife;

  @media (max-width: 500px) {
    padding-right: 60px;
  }
`;

const Transfer = styled.a`
  position: absolute;
  left: 176px;
  color: #3eacdc;
  text-decoration: none;
  font-family: sans-serife;

  @media (max-width: 500px) {
    left: 150px;
  }
`;

export { Button, InputFormWrapper, Link, Error, Transfer };
