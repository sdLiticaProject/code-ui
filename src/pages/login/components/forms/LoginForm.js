import React from 'react';
import {useForm} from 'react-hook-form';
import {NavLink, useHistory} from 'react-router-dom';
import Snackbar from '@material-ui/core/Snackbar';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import {useSelector} from 'react-redux';
import useActions from '../../../../hooks/useAction';
import {HOME, LOGIN} from '../../../../constants/routes';
import {
  loginUser,
  LOGIN_FAIL,
  LOGIN_LOADING,
  LOGIN_SUCCESS
} from '../../../../actions/loginActions';
import {
  Button,
  InputFormWrapper,
  LoadIndicator,
  NavLinkWrapper,
  Error,
  Transfer,
  FormWrapper
} from './FormsStyles';

function LoginForm() {
  const {handleSubmit, register, errors} = useForm();
  const [submitAction] = useActions([loginUser]);
  const history = useHistory();
  const [isSnackbarOpen, setSnackbarOpen] = React.useState(false);
  const [snackMessage, setSnackMessage] = React.useState('');
  const isLoginLoading = useSelector(state => state.login.type) === LOGIN_LOADING;

  const handleSnackbarClose = (event, reason) => {
    if (reason === 'clickaway') {
      return;
    }

    setSnackbarOpen(false);
  };

  const submit = data => {
    if (data.email !== '' && data.password !== '') {
      submitAction(data.email, data.password).then(e => {
        if (e.type && e.type === LOGIN_FAIL) {
          setSnackMessage(e.message);
          setSnackbarOpen(true);
        } else if (e.type && e.type === LOGIN_SUCCESS) {
          history.push(HOME);
        }
      });
    }
  };

  return (
    <FormWrapper onSubmit={handleSubmit(submit)}>
      <InputFormWrapper>
        <label htmlFor="email">Email</label>
        <input
          ref={register({
            required: 'required',
            pattern: {
              value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
              message: 'invalid email address'
            }
          })}
          name="email"
        />
        <Error>{errors.email && errors.email.message}</Error>
      </InputFormWrapper>
      <br />
      <InputFormWrapper>
        <label htmlFor="password">Password</label>
        <input
          ref={register({
            required: 'required',
            pattern: {
              value: /^[A-Z0-9._]{3,15}$/i,
              message: 'invalid password'
            }
          })}
          type="password"
          name="password"
          id="password"
        />
        <Error>{errors.password && errors.password.message}</Error>
      </InputFormWrapper>
      {/* <NavLink to={`${LOGIN}?tab=restore`}>
        <Link>Forgot password</Link>
      </NavLink> */}
      <br />
      <Button type="submit">Log in</Button>
      {isLoginLoading && <LoadIndicator />}
      <NavLinkWrapper>
        <NavLink to={`${LOGIN}?tab=reg`}>
          <Transfer>Need an account?</Transfer>
        </NavLink>
      </NavLinkWrapper>
      <Snackbar
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'center'
        }}
        open={isSnackbarOpen}
        autoHideDuration={6000}
        onClose={handleSnackbarClose}
        message={snackMessage}
        action={
          <React.Fragment>
            <IconButton
              size="small"
              aria-label="close"
              color="inherit"
              onClick={handleSnackbarClose}>
              <CloseIcon fontSize="small" />
            </IconButton>
          </React.Fragment>
        }
      />
    </FormWrapper>
  );
}

export default LoginForm;
