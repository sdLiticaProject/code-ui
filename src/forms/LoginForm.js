import React, { useState } from "react";
import { useForm } from "react-hook-form";
import { NavLink } from "react-router-dom";
import { REGISTRATION, RECOVERY } from "../constants/routes";
import { loginUser } from "../actions/loginActions";
import useActions from "../hooks/useAction";
import { Button, InputFormWrapper, Link, Error, Transfer } from "./FormsStyles";

function LoginForm() {
  const [login, setLogin] = useState("");
  const [password, setPassword] = useState("");
  const { handleSubmit, register, errors } = useForm();

  //Alternative bindActionCreators
  const [submitAction] = useActions([loginUser]);

  const submit = () => {
    if (login !== "" && password !== "") {
      submitAction(login, password);
    }
  };

  return (
    <form onSubmit={handleSubmit(submit)}>
      <InputFormWrapper>
        <label htmlFor="login">Email</label>
        <input
          ref={register({
            required: "required",
            pattern: {
              value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
              message: "invalid email address",
            },
          })}
          name="login"
          onChange={(e) => setLogin(e.target.value)}
        />
        <Error>{errors.login && errors.login.message}</Error>
      </InputFormWrapper>
      <br />
      <InputFormWrapper>
        <label for="password">Password</label>
        <input
          ref={register({
            required: "required",
            pattern: {
              value: /^[A-Z0-9._]{6,15}$/i,
              message: "invalid password",
            },
          })}
          type="password"
          name="password"
          onChange={(e) => setPassword(e.target.value)}
        />
        <Error>{errors.password && errors.password.message}</Error>
      </InputFormWrapper>
      <NavLink to={RECOVERY}>
        <Link>Forgot password</Link>
      </NavLink>
      <br />
      <Button type="submit">Log in</Button>

      <NavLink to={REGISTRATION}>
        <Transfer>Need an account?</Transfer>
      </NavLink>
    </form>
  );
}

export default LoginForm;
