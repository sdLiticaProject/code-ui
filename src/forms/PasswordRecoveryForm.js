import React, {useState} from 'react';
import {NavLink} from 'react-router-dom';
import {useForm} from 'react-hook-form';
import {LOGIN} from '../constants/routes';
import {loginUser} from '../actions/loginActions';
import useActions from '../hooks/useAction';
import {Button, InputFormWrapper, Error, Link} from './FormsStyles';

function PasswordRecoveryForm() {
  const [email, setEmail] = useState('');
  const {handleSubmit, register, errors} = useForm();

  // Alternative bindActionCreators
  const [submitAction] = useActions([loginUser]);

  const submit = () => {
    if (email !== '') {
      submitAction(email);
    }
  };

  return (
    <form onSubmit={handleSubmit(submit)}>
      <InputFormWrapper>
        <label htmlFor="email">Enter your email to recovery password</label>
        <input
          ref={register({
            required: 'required',
            pattern: {
              value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
              message: 'invalid email address'
            }
          })}
          name="email"
          onChange={e => setEmail(e.target.value)}
        />
        <Error>{errors.email && errors.email.message}</Error>
      </InputFormWrapper>
      <br />
      <NavLink to={LOGIN}>
        <Link>Back to authorization</Link>
      </NavLink>
      <br />
      <Button type="submit">Get the code</Button>
    </form>
  );
}

export default PasswordRecoveryForm;
