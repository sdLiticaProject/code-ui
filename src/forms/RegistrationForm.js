import React, {useState} from 'react';
import {NavLink} from 'react-router-dom';
import {useForm} from 'react-hook-form';
import {LOGIN} from '../constants/routes';
import {loginUser} from '../actions/loginActions';
import useActions from '../hooks/useAction';
import {Button, InputFormWrapper, Error, Transfer} from './FormsStyles';

function RegistrationForm() {
  const [login, setLogin] = useState('');
  const [password, setPassword] = useState('');
  const {handleSubmit, register, errors} = useForm();

  // Alternative bindActionCreators
  const [submitAction] = useActions([loginUser]);

  const submit = () => {
    if (login !== '' && password !== '') {
      submitAction(login, password);
    }
  };

  return (
    <form onSubmit={handleSubmit(submit)}>
      <InputFormWrapper>
        <label htmlFor="login">Enter your email</label>
        <input
          ref={register({
            required: 'required',
            pattern: {
              value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
              message: 'invalid email address'
            }
          })}
          name="login"
          onChange={e => setLogin(e.target.value)}
        />
        <Error>{errors.login && errors.login.message}</Error>
      </InputFormWrapper>
      <br />
      <InputFormWrapper>
        <label htmlFor="password">Create the password(6-15 symbols)</label>
        <input
          ref={register({
            required: 'required',
            pattern: {
              value: /^[A-Z0-9._]{6,15}$/i,
              message: 'invalid password'
            }
          })}
          type="password"
          name="password"
          onChange={e => setPassword(e.target.value)}
        />
        <Error>{errors.password && errors.password.message}</Error>
      </InputFormWrapper>
      <br />
      <Button type="submit">Sign up</Button>
      <NavLink to={LOGIN}>
        <Transfer>Have an account?</Transfer>
      </NavLink>
    </form>
  );
}

export default RegistrationForm;
