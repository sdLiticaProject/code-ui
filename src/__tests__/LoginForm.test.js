import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {mount} from 'enzyme';
import {Button} from '../forms/FormsStyles';
import CustomBrowserRouter from '../helpers/CustomBrowserRouter';
import LoginForm from '../forms/LoginForm';
import { beforeEach, describe, expect, it } from "@jest/globals";

const defaultStore = {responsive: {fakeWidth: 1200}};
const mockedStore = configureStore()(defaultStore);

export const mountWithProvider = (children) => (store = mockedStore) =>
  mount(
    <Provider store={store}>
      <CustomBrowserRouter>{children}</CustomBrowserRouter>
    </Provider>
  );

const mockStore = configureStore([]);

describe('Testing LoginForm component', () => {
  let store;
  let component;
  let mountCmp;

  beforeEach(() => {
    store = mockStore({
      myState: 'sample text',
    });

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <CustomBrowserRouter>
          <LoginForm />
        </CustomBrowserRouter>
      </Provider>
    );

    mountCmp = mountWithProvider(<LoginForm />)();
  });

  it('component renders correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('LoginForm exists as connected mounted component', () => {
    expect(mountCmp.find(LoginForm).length).toBe(1);
  });

  it('submit button exists', () => {
    expect(mountCmp.find(Button).length).toBe(1);
  });

  it('button should have submit type', () => {
    expect(component.root.findByType(Button).props.type).toBe('submit');
  });
});
