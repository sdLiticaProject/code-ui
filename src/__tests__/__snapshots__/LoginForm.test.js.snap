// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoginForm component component renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="sc-AxjAm cVoBpx"
  >
    <label
      htmlFor="login"
    >
      Email
    </label>
    <input
      name="login"
      onChange={[Function]}
    />
    <text
      className="sc-AxiKw cTDmNm"
    />
  </div>
  <br />
  <div
    className="sc-AxjAm cVoBpx"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      name="password"
      onChange={[Function]}
      type="password"
    />
    <text
      className="sc-AxiKw cTDmNm"
    />
  </div>
  <a
    aria-current={null}
    href="/recovery"
    onClick={[Function]}
  >
    <a
      className="sc-AxhCb hMurPe"
    >
      Forgot password
    </a>
  </a>
  <br />
  <button
    className="sc-AxirZ dygZym"
    type="submit"
  >
    Log in
  </button>
  <a
    aria-current={null}
    href="/registr"
    onClick={[Function]}
  >
    <a
      className="sc-AxhUy hvhAfX"
    >
      Need an account?
    </a>
  </a>
</form>
`;
